### PR TITLE
Refactor rasterio usage to improve import time

### PR DESCRIPTION
**Pull request by djhoese**
As described by @simonrp84 in https://github.com/pytroll/satpy/issues/2159, the import time of Satpy is getting slower. Most of this is due to the dependencies being imported like xarray, rasterio, and dask. This PR is an attempt at helping that along by moving and delaying rasterio import in trollimage's XRImage until it is actually used (in `.rio_save`). The code that was there wasn't pretty, but I did the best I could without rewriting everything.

In my tests by doing:

```
python -X importtime -c "from trollimage.xrimage import XRImage"
```

I saw an import time go from ~715ms to ~609ms. After rerunning it a couple times I think the newer time is closer to 647ms though. This is not as big of an increase as I was hoping. For the record, xrimage takes 296ms to import.

 - [ ] Closes #xxxx (remove if there is no corresponding issue, which should only be the case for minor changes)
 - [ ] Tests added (for all bug fixes or enhancements)
 - [ ] Tests passed (for all non-documentation changes)
 - [ ] Passes ``git diff origin/master **/*py | flake8 --diff`` (remove if you did not edit any Python files)
 - [ ] Fully documented (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

**Comment by djhoese** [comment]
FYI rerunning `main` importtime I see anywhere from 750ms to 840ms so maybe this is a decent improvement.

File: trollimage/_xrimage_rasterio.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    from rasterio.enums import Resampling
+except ImportError:
+    rasterio = None
+
+try:
+    # rasterio 1.0+
+    from rasterio.windows import Window
+except ImportError:
+    # raster 0.36.0
+    # remove this once rasterio 1.0+ is officially available
```
Comment (mraspaud) [inline trollimage/_xrimage_rasterio.py:34]: It is now :) should we use the opportunity?

**Review by mraspaud — APPROVED** [review]
Looks good, thanks for the refactoring! just one tiny comment, maybe it's worth the clean up?

File: trollimage/_xrimage_rasterio.py (hunk 1 above)
Comment (djhoese) [reply trollimage/_xrimage_rasterio.py:34]: Looks like 1.0.0 came out 4 years ago so I think we're good. I'll remove it. Unless we have a limit on the setup.py already I don't think I need to add one...oh maybe its an extra.

**Pull request merged**